tests/foreman/cli/test_discoveryrule.py

```diff
@@ -132,6 +132,9 @@ def test_positive_create_with_search(self, discoveryrule_factory):
         search_query = 'cpu_count = 2'
         rule = discoveryrule_factory(options={'search': search_query})
         assert rule.search == search_query
+        custom_query = 'processor = x86'
+        rule = discoveryrule_factory(options={'search': custom_query})
```

Comment:
Can we move `discoveryrule_factory` to cli_factory ?

Reply:
We can but I think the factory is more of a place for generic fixtures so better to keep it here.

Reply:
first cli_factory aren't fixtures, and second we don't have any restrictions on helper functions to keep generic, component specific helpers can be moved to it as well and in future can be used in other components too.

Reply:
I would still keep this here